client/FxAccountClient.js

```diff
@@ -913,9 +955,21 @@ define([
           )
         );
 
-        return self.request.send('/password/change/finish', 'POST', hawkCreds, {
+        var queryParams = '';
+        if (options.keys) {
+          queryParams = '?keys=true';
+        }
+
+        return self.request.send('/password/change/finish' + queryParams, 'POST', hawkCreds, {
           wrapKb: newWrapKb,
-          authPW: sjcl.codec.hex.fromBits(newCreds.authPW)
+          authPW: sjcl.codec.hex.fromBits(newCreds.authPW),
+          sessionToken: options.sessionToken
+        })
+        .then(function (accountData) {
+          if (options && options.keys && accountData.keyFetchToken) {
```

Comment:
nit: looks like `options` is defaulted to {}, could probably remove `options &&`.